tests/unit/test_utils.py

```diff
@@ -52,3 +64,62 @@ def test_parse_memory_string(the_str: str, expected: float) -> None:
 def test_parse_incorrect_memory_string(the_str: str) -> None:
     with pytest.raises(ValueError):
         parse_memory_string(the_str)
+
+
+WDirCheck = Callable[[Path], Namespace]
+
+
+@pytest.fixture()
+def w_dir_check() -> WDirCheck:
+    def _func(path: Path) -> Namespace:
+        action = WriteEnabledDirectoryAction(option_strings=["-m"], dest="d")
+        parser = create_autospec(ArgumentParser)
+
+        def error(message: str) -> NoReturn:
+            raise ValueError(message)
+
+        parser.error.side_effect = error
+        namespace = Namespace()
+        action(parser, namespace, str(path))
+        return namespace
+
+    return _func
+
+
+def test_write_enabled_dir_ok(w_dir_check: WDirCheck, tmp_path: Path) -> None:
```

Comment:
These tests are doing filesystem stuff, can you move these to the tests directory (technically they are not `unit` tests). I don't feel very strongly about this though :)

Reply:
Depends on your definition of the unit test. In my book, a unit test is a test that runs without external dependencies within the same python interpreter (already available on the machine). So in that sense accessing the filesystem is not an integration test. Otherwise all tests here would be integration tests because they write to the FS the test file (see makepyfile).

Reply:
> Depends on your definition of the unit test.

In my book, unit tests don't interact with the filesystem or usingg system resources like threads and whatnot.

> Otherwise, all tests here would be integration tests because they write to the FS the test file (see makepyfile).

Yeah, that's true :)

Reply:
We can do a follow-up PR that removes the `unit` layer then?